terminal-emulator/src/main/java/com/termux/terminal/TerminalEmulator.java

```diff
@@ -0,0 +1,2447 @@
+        }
+
+        resizeScreen();
+    }
+
+    private void resizeScreen() {
+        final int[] cursor = {mCursorCol, mCursorRow};
+        int newTotalRows = (mScreen == mAltBuffer) ? mRows : mMainBuffer.mTotalRows;
+        mScreen.resize(mColumns, mRows, newTotalRows, cursor, getStyle(), isAlternateBufferActive());
+        mCursorCol = cursor[0];
+        mCursorRow = cursor[1];
+    }
+
+    public int getCursorRow() {
+        return mCursorRow;
+    }
+
+    public int getCursorCol() {
+        return mCursorCol;
+    }
+
+    /** Get the terminal cursor style. It will be one of {@link #TERMINAL_CURSOR_STYLES_LIST} */
+    public int getCursorStyle() {
+        return mCursorStyle;
+    }
+
+    /** Set the terminal cursor style. */
+    public void setCursorStyle() {
+        Integer cursorStyle = null;
+
+        if (mClient != null)
+            cursorStyle = mClient.getTerminalCursorStyle();
+
+        if (cursorStyle == null || !Arrays.asList(TERMINAL_CURSOR_STYLES_LIST).contains(cursorStyle))
+            mCursorStyle = DEFAULT_TERMINAL_CURSOR_STYLE;
+        else
+            mCursorStyle = cursorStyle;
+    }
+
+    public boolean isReverseVideo() {
+        return isDecsetInternalBitSet(DECSET_BIT_REVERSE_VIDEO);
+    }
+
+
+
+    public boolean isCursorEnabled() {
+        return isDecsetInternalBitSet(DECSET_BIT_CURSOR_ENABLED);
+    }
+    public boolean shouldCursorBeVisible() {
+        if (!isCursorEnabled())
+            return false;
+        else
+            return mCursorBlinkingEnabled ? mCursorBlinkState : true;
+    }
+
+    public void setCursorBlinkingEnabled(boolean cursorBlinkingEnabled) {
+        this.mCursorBlinkingEnabled = cursorBlinkingEnabled;
+    }
+
+    public void setCursorBlinkState(boolean cursorBlinkState) {
+        this.mCursorBlinkState = cursorBlinkState;
+    }
+
+
+
+    public boolean isKeypadApplicationMode() {
+        return isDecsetInternalBitSet(DECSET_BIT_APPLICATION_KEYPAD);
+    }
+
+    public boolean isCursorKeysApplicationMode() {
+        return isDecsetInternalBitSet(DECSET_BIT_APPLICATION_CURSOR_KEYS);
+    }
+
+    /** If mouse events are being sent as escape codes to the terminal. */
+    public boolean isMouseTrackingActive() {
+        return isDecsetInternalBitSet(DECSET_BIT_MOUSE_TRACKING_PRESS_RELEASE) || isDecsetInternalBitSet(DECSET_BIT_MOUSE_TRACKING_BUTTON_EVENT);
+    }
+
+    private void setDefaultTabStops() {
+        for (int i = 0; i < mColumns; i++)
+            mTabStop[i] = (i & 7) == 0 && i != 0;
+    }
+
+    /**
+     * Accept bytes (typically from the pseudo-teletype) and process them.
+     *
+     * @param buffer a byte array containing the bytes to be processed
+     * @param length the number of bytes in the array to process
+     */
+    public void append(byte[] buffer, int length) {
+        for (int i = 0; i < length; i++)
+            processByte(buffer[i]);
+    }
+
+    private void processByte(byte byteToProcess) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.